api/lib/domain/read-models/CampaignAssessmentParticipationSummary.js

```diff
@@ -0,0 +1,55 @@
+const Assessment = require('../models/Assessment');
+
+const statuses = {
+  SHARED: 'shared',
+  ONGOING: 'ongoing',
+  COMPLETED: 'completed',
+};
+
+class CampaignAssessmentParticipationSummary {
+  constructor({
+    campaignParticipationId,
+    // attributes
+    userId,
+    firstName,
+    lastName,
+    participantExternalId,
+    sharedAt,
+    state,
+    isShared,
+    targetedSkillCount,
+    validatedTargetedSkillCount,
+  } = {}) {
+    this.campaignParticipationId = campaignParticipationId;
+    // attributes
+    this.userId = userId;
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.participantExternalId = participantExternalId;
+    this.sharedAt = sharedAt;
+    this.status = this._getStatus(isShared, state);
+    this.targetedSkillCount = targetedSkillCount;
+    this.validatedTargetedSkillCount = validatedTargetedSkillCount;
+    if (isShared) {
+      this._computeMasteryPercentage();
+    }
+  }
+
+  _computeMasteryPercentage() {
+    if (this.targetedSkillCount !== 0) {
+      this.masteryPercentage = Math.round(this.validatedTargetedSkillCount * 100 / this.targetedSkillCount);
+    } else {
+      this.masteryPercentage = 0;
+    }
+  }
+
+  _getStatus(isShared, state) {
+    if (isShared) return CampaignAssessmentParticipationSummary.statuses.SHARED;
+    if (state === Assessment.states.COMPLETED) return CampaignAssessmentParticipationSummary.statuses.COMPLETED;
+    return CampaignAssessmentParticipationSummary.statuses.ONGOING;
+  }
+}
```

Comment:
Pas convaincu pas le fait d'avoir ça ici et pas au moment de la construction du read-model. 
Le read-model est censé être un objet plat directement serializable. (ça veut pas dire qu'on ne teste pas cette logique ailleurs).

Reply:
Voir ici : http://gorodinski.com/blog/2012/04/25/read-models-as-a-tactical-pattern-in-domain-driven-design-ddd/
```
public OrderReadModel
{
  public string Id { get; private set; }
  public string OrderNumber { get; private set; }
  public DateTime Date { get; private set; }
  public decimal Total { get; private set; }
} 
```
> Note that this class contains no behavior and is read only.

Reply:
On est toujours dans le cas d'un read-model (en read-only).
Ce sont juste des données "dérivées" par rapport aux données d'entrée.

Reply:
Vu avec @EGI-OCTO :
Le  read-model objet peut avoir de la logique de construction, mais ton objet doit être plat une fois que la construction est faite 
i.e. Quand l'objet est crée je peux le sérializer tout de suite, je n'ai pas besoin de faire de calculs supplémentaires